src/cryptonote_core/oxen_name_system.cpp

```diff
@@ -100,26 +103,27 @@ std::pair<std::basic_string_view<unsigned char>, std::basic_string_view<unsigned
   return result;
 }
 
-std::string lns::mapping_value::to_readable_value(cryptonote::network_type nettype, lns::mapping_type type) const
+std::string ons::mapping_value::to_readable_value(cryptonote::network_type nettype, ons::mapping_type type) const
 {
   std::string result;
   if (is_lokinet_type(type))
   {
     result = oxenmq::to_base32z(to_view()) + ".loki";
-  }
-  else if (type == lns::mapping_type::wallet)
-  {
-    cryptonote::address_parse_info addr_info = {};
-    if (len == sizeof(addr_info))
-    {
-      std::memcpy(&addr_info, buffer.data(), len);
-      result = cryptonote::get_account_address_as_str(nettype, addr_info.is_subaddress, addr_info.address);
+  } else if (type == ons::mapping_type::wallet) {
+    cryptonote::address_parse_info addr_info{0};
+    auto iter = std::next(buffer.begin(),1);
+    std::memcpy(&addr_info.address.m_spend_public_key.data, &*iter, 32);
+    std::advance(iter,32);
+    std::memcpy(&addr_info.address.m_view_public_key.data, &*iter, 32);
```

Comment:
```suggestion
    auto* bufpos = &buffer[1];
    std::memcpy(addr_info.address.m_spend_public_key.data, bufpos, 32);
    bufpos += 32;
    std::memcpy(addr_info.address.m_view_public_key.data, bufpos, 32);
```
It's simpler to just do pointer arithmetic here rather than try to use iterators.

Reply:
Also you don't need the & on the destinations here because the `.data` field is already a C array, which is already a pointer.